desktop-widgets/subsurfacewebservices.cpp

```diff
@@ -126,6 +126,7 @@ bool DivelogsDeWebServices::prepare_dives_for_divelogs(const QString &tempfile,
 			goto error_close_zip;
 		}
 		free((void *)membuf);
+		free_buffer(&mb);
```

Comment:
the `mb` variable itself is allocated on the stack.
the above call already handles the memory de-allocation - `free((void *)membuf);`
but i would remove the `free((void *)membuf);` line and leave the line you've added `free_buffer(&mb);`